### ISSUE TITLE
Renaming a container with the same name as its current name

### ISSUE THREAD

**Issue by zffocussss** [docker/compose#6704]
## Description of the issue
I use docker-compose to running service with a docker-compose.yml.
(python service,mongo,redis)
everytime python codes are updated,it will be run.
docker-compose build --build-arg
docker-compose up -d --remove-orphans
## Context information (for bug reports)

**Output of `docker-compose version`**
```
docker-compose version 1.23.2, build 1110ad01

```

**Output of `docker version`**
```
Client:
 Version:           18.09.2
 API version:       1.39
 Go version:        go1.10.6
 Git commit:        6247962
 Built:             Sun Feb 10 04:13:50 2019
 OS/Arch:           linux/amd64
 Experimental:      false

Server: Docker Engine - Community
 Engine:
  Version:          18.09.2
  API version:      1.39 (minimum version 1.12)
  Go version:       go1.10.6
  Git commit:       6247962
  Built:            Sun Feb 10 03:42:13 2019
  OS/Arch:          linux/amd64
  Experimental:     false
```

**Output of `docker-compose config`**
(Make sure to add the relevant `-f` and other flags)
```
(paste here)
```

## Steps to reproduce the issue

this error disappeared when I rerun the same process (I use bitbucket to do this)

### Stacktrace / full error message

```
I am running a service and have a container name (abc).

ERROR: for abc  Renaming a container with the same name as its current name

```

## Additional information

Host Ubuntu 16.04

**Comment by zffocussss** [comment]
<img width="615" alt="WeChat8099da8f49bc916b9a27d3d0ce5bd727" src="https://user-images.githubusercontent.com/46145695/57679616-56fdfc80-765e-11e9-9494-3357425200cf.png">

**Comment by zffocussss** [comment]
It's my first time to meet this error.Can somebody help me?

**Comment by jcsirot** [comment]
Hello @zffocussss Could you post a compose file and the list of commands which for which you were able to reproduce the issue?

**Comment by NicolasDorier** [comment]
just want to report, got same error once on same docker-compose version.
It happened once on many deployment, I don't know if I will be able to reproduce, down and up fixed the situation.

**Comment by NicolasDorier** [comment]
```
Docker version 18.09.5, build e8ff056
docker-compose version 1.23.2, build 1110ad01
```

**Comment by NicolasDorier** [comment]
Additional info: I did not do `build`, it was just a docker-compose up.

**Comment by kai-chu** [comment]
I met the problem as well when I try to run two commands docker-compose up in parallel.
docker-compose version 1.21.0, build 5920eb0

**Comment by uptownhr** [comment]
I have also started to experience this. I have not changed anything though. Just started happening.

**Comment by mmathys** [comment]
I have this issue

**Comment by sandu-c** [comment]
me too

**Comment by mmathys** [comment]
cc @jcsirot

**Comment by douglasg14b** [comment]
And same here....

**Comment by aminvakil** [comment]
See https://github.com/getsentry/self-hosted/runs/4921403940?check_suite_focus=true#step:4:2520

**Comment by chadwhitacre** [comment]
@ndeloof @ulyssessouza Could we reopen this ticket? Seems to still be an issue. 🙏

**Comment by ndeloof** [comment]
Does this issue also affect Compose v2?

**Comment by aesyondu** [comment]
> Same problem here
> 
> docker-compose down and then up helped.

Thanks! Confirm that this works. My guess is that this happens when:
- `docker-compose up` is executed, and then
- `docker-compose.yml` file is modified, and then
- `docker-compose up` is executed again.

**Comment by SubsterNL** [comment]
I'm getting this error a lot when trying to edit the existing container then deploying it again (and choosing yes to automatically replace it on the dialog).

**Comment by Veetaha** [comment]
I didn't modify the `docker-compose.yml` between the reruns. I only ran `docker compose up` while the stack is already running (I only changed my Rust code sources, so docker build should've happened only), but I got this error.

**Comment by ndeloof** [comment]
I tried to reproduce this issue with:
compose.yaml
```yaml
services:
  test:
    build: .
    container_name: toto
```

Dockerfile
```
FROM nginx
```

```
$ docker compose up --build -d
[+] Building 0.1s (5/5) FINISHED                                                
 => [internal] load build definition from Dockerfile                       0.0s
 => => transferring dockerfile: 60B                                        0.0s
...
 => => naming to docker.io/library/truc-test                               0.0s
[+] Running 1/1
 ⠿ Container toto  Started     
$ docker ps | grep toto
967d276e1fb9   truc-test   "/docker-entrypoint.…"   About a minute ago   Up About a minute   80/tcp, 8080/tcp   toto
```

Then update Dockerfile to add `EXPOSE 8080`, and ran up again:
```
$ docker compose --ansi=never up --build -d
[+] Building 0.1s (5/5) FINISHED                                                
 => [internal] load build definition from Dockerfile                       0.0s
 => => transferring dockerfile: 60B                                        0.0s
 => [internal] load .dockerignore                                          0.0s
 => => transferring context: 2B                                            0.0s
 => [internal] load metadata for docker.io/library/nginx:latest            0.0s
 => CACHED [1/1] FROM docker.io/library/nginx                              0.0s
 => exporting to image                                                     0.0s
 => => exporting layers                                                    0.0s
 => => writing image sha256:b329c29666ceb98ccd2f08cd3b450b046da3e0e5be193  0.0s
 => => naming to docker.io/library/truc-test                               0.0s
Container toto  Recreate
Container toto  Recreated
Container toto  Starting
Container toto  Started
```

Compose (at least, latest v2 codebase) doesn't seem to have issue recreating container, even with a name set by configuration.
Could you please offer a reproduction scenario ?

**Comment by ndeloof** [comment]
Closing this issue as inactive/obsolete

**Comment by isaacsolo** [comment]
me as well 😮‍💨

**Comment by ndeloof** [comment]
@isaacsolo please create a follow-up issue with more details
